storybooknative/src/Text/index.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+import * as React from "react";
+import { Text as RNText } from "react-native";
+
+import styled from "../styled";
+import defaultTokens from "../defaultTokens";
+import { TYPE_OPTIONS, WEIGHT_OPTIONS, ALIGN_OPTIONS, SIZE_OPTIONS } from "./consts";
+
+import type { Props } from "./index";
+
+const getTypeToken = ({ theme, type }) => {
+  const typeTokens = {
+    [TYPE_OPTIONS.PRIMARY]: theme.orbit.colorTextPrimary,
+    [TYPE_OPTIONS.SECONDARY]: theme.orbit.colorTextSecondary,
+    [TYPE_OPTIONS.ATTENTION]: theme.orbit.colorTextAttention,
+    [TYPE_OPTIONS.INFO]: theme.orbit.colorTextInfo,
+    [TYPE_OPTIONS.SUCCESS]: theme.orbit.colorTextSuccess,
+    [TYPE_OPTIONS.WARNING]: theme.orbit.colorTextWarning,
+    [TYPE_OPTIONS.CRITICAL]: theme.orbit.colorTextCritical,
+    [TYPE_OPTIONS.WHITE]: theme.orbit.colorTextWhite,
+  };
+  return typeTokens[type];
+};
+
+const getWeightToken = ({ theme, weight }) => {
+  const weightTokens = {
+    [WEIGHT_OPTIONS.NORMAL]: theme.orbit.fontWeightNormal,
+    [WEIGHT_OPTIONS.BOLD]: theme.orbit.fontWeightBold,
+  };
+  return weightTokens[weight];
+};
+
+const getSizeToken = ({ theme, size }) => {
+  const sizeTokens = {
+    [SIZE_OPTIONS.LARGE]: theme.orbit.fontSizeTextLarge,
+    [SIZE_OPTIONS.NORMAL]: theme.orbit.fontSizeTextNormal,
+    [SIZE_OPTIONS.SMALL]: theme.orbit.fontSizeTextSmall,
+  };
+  return sizeTokens[size];
+};
+
+const StyledText = styled(RNText, props => ({
+  fontSize: getSizeToken(props),
+  fontWeight: getWeightToken(props),
+  color: getTypeToken(props),
+  lineHeight: props.theme.orbit.lineHeightText,
+  textAlign: props.align,
+  textTransform: props.uppercase ? `uppercase` : `none`,
+  fontStyle: props.italic ? `italic` : `normal`,
+  margin: 0,
+}));
+
+StyledText.defaultProps = {
+  theme: {
+    orbit: {
```

Comment:
Yeah, I see.

Looking into the test of this component, the only thing which is tested is props passing.
I would rather reduce the test only to one component (renderer), instead of a ton of different combinations, and would use only one snapshot with full `theme` prop.

Also, is there any chance to test visual styles with `jest-styled-component` and `expect(component).toHaveStyleRule("property", "value")`? We just start to use it and we wanna have it in all tests in the future.

See [this](https://github.com/kiwicom/orbit-components/blob/b90e88dbdec8d83db4259065121b553872917f33/src/Icon/__tests__/index.test.js#L21).

Reply:
Oops, I fixed it already. How do you revert?

Reply:
It's fine, leave as it is for now. We need to discuss of how we want to move forward.

Reply:
We will try to incorporate `jest-styled-component` testing with `toHaveStyleRule` in a different PR. It's already becoming too big, and not just about the Text component.